basic_module/orderer/src/core.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::types::{OrdererError, TxMetadata};
+
+pub trait Orderer {
+    fn filter_and_order_transactions(metadata_list: Vec<TxMetadata>) -> Result<Vec<usize>, OrdererError>;
```

Comment:
&self

Reply:
So, each handle must have a dummy trait and a dummy self reference, even if they are not used?

Reply:
What do you mean by 'dummy trait'?

Reply:
The current Orderer trait is a dummy trait. 
It doesn't use `self`, so it doesn't have to be a trait at all.

If there is some way of registering a global function as a callback, we don't need the Orderer module at all.